### DISCUSSION TITLE
Simultaneous Bi-Directional Transmission of video and telemetry

### DISCUSSION THREAD

**Discussion by Mas313** [svpcom/wfb-ng#172] [General]
I wanted to know if your developed wifibroadcast using CCA or CSMA/CA channel access method to transmit RC/mavlink commands from ground station and video from air unit? Also you have mentioned concept of radio ports in one of your discussions/issues related to transmission of multiple streams of video, RC and bi-directional mavlink. So what actually is a radio port?. Does each radio port use different channel (frequency) to transmit.

Also i want to know that at ground side if there is any interference than how receive code extracts transmitted packets from air unit and not from wifi signal on the same frequency, since it is receiving packets from air unit and from local wifi source (with good RSSI).

**Comment by svpcom** [comment]
1. WFB use standard ieee802.11 radio to detect that medium is busy, but doesn't use any collision avoidance methods. I.e. it try to inject packets and wifi radio will check if medium is free then transmit.

2. Radio "ports" is  different MAC addresses which WFB use to separate different data streams. No - the same frequency is used because standard wifi hardware unable to listen on multiple freqs simultaneously.

3. If two packets interfere each other then radio will unable to decode both of them. In case of packet loss WFB use FEC (Reed-Solomon codes) to recover some percent of missed packets.  Input and output packets use different ports (MAC addressed) and host will not receive its own packets. To distinguish WFB traffic from ordinary wifi packets special MAC addresses and data encryption are used

**Comment by Mas313** [comment]
Thanks for the detailed response, it really made me understand well, i want your expert advice on 2.4 Ghz atheros 9271 based wifi dongles. I have seen in monitor mode if i inject packet it gives me data rate of 1 Mbps (DSSS).   Do you know how can i change its data rate to 2 Mbps or 3 Mbps. As per 802.11b specifications, preamble and sync words are sent at 1 Mbps whereas data payload can be sent with different bit rate (1 Mbps to 11 Mbps). However, i am getting 1 Mbps for data rate. I have downloaded its firmware but not able to figure out how force packet injection with 802.11b with data rate  greater than 1 Mbps.  I have run following commands but received errors
```
sudo iwconfig wlan0 rate 6M
sudo iw dev wlan0 set bitrates legacy-5 6
```

**Reply by svpcom** [reply]
I didn't tried atheros cards but I know that there are patched opensource firmware with advanced injection features for them. Try to see ez-wifibroadcast and/or OpenHD repos

**Comment by Mas313** [comment]
Thanks for the reply.

**Comment by Mauronic** [comment]
@svpcom That was a very helpful answer. Why was collision avoidance not implemented? I’m curious because my application requires 2 ground vehicles to work indoors alongside other WiFi devices.

**Reply by svpcom** [reply]
For indoor and crowded radio environment use normal wifi.  wfb-ng is designed for long-range links and use less efficient collision resolution method but it works for long distance links compared to ACKs

**Reply by Mauronic** [reply]
thanks for getting back to me :)I haven't been able to find a low-ish (100ms) latency HD FPV video system that is (or close to) 802.11 compliant. Are you aware of one?

**Reply by svpcom** [reply]
wfb-ng with openipc cameras can provide 40ms glass-to-glass

**Reply by Mauronic** [reply]
Impressive, but I am asking if you have seen a solution that uses "normal wifi" for my indoor and crowded radio environment.

**Reply by svpcom** [reply]
Use the same openipc camera + low-latency player (radxa zero 3w/e + 144Hz monitor) over "normal wifi". You will get the same latency

**Reply by Mauronic** [reply]
That's great news, I will try it! I am very curious to see the G2G latency for radxa zero 3w/e + 144Hz monitor vs mobile phone vs Windows PC
